bundles/org.openhab.binding.mynice/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,165 @@
+				<label>Hostname</label>
+				<description>Hostname or IP address of the device</description>
+			</parameter>
+			<parameter name="macAddress" type="text" pattern="([0-9A-Fa-f]{2}:){5}[0-9A-Fa-f]{2}(\s*#.*)*"
+				required="true">
+				<label>MAC Address</label>
+				<description>The MAC address of the IT4Wifi</description>
+			</parameter>
+			<parameter name="password" type="text">
+				<context>password</context>
+				<label>Pairing Key</label>
+				<description>Pairing Key needed to access the device, provided by the bridge itself</description>
+			</parameter>
+		</config-description>
+	</bridge-type>
+
+	<thing-type id="swing">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="it4wifi"/>
+		</supported-bridge-type-refs>
+
+		<label>Swing Gate</label>
+		<description>A double wing gate</description>
+
+		<channels>
+			<channel id="status" typeId="doorstatus"/>
+			<channel id="obstruct" typeId="obstruct"/>
+			<channel id="moving" typeId="moving"/>
+			<channel id="command" typeId="command"/>
+			<channel id="t4command" typeId="t4command"/>
+		</channels>
+
+		<representation-property>id</representation-property>
+
+		<config-description>
+			<parameter name="id" type="text" required="true">
+				<label>ID</label>
+				<description>ID of the gate on the TP4 bus connected to the bridge</description>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<thing-type id="sliding">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="it4wifi"/>
+		</supported-bridge-type-refs>
+
+		<label>Sliding Gate</label>
+		<description>A sliding gate</description>
+
+		<channels>
+			<channel id="status" typeId="doorstatus"/>
+			<channel id="obstruct" typeId="obstruct"/>
+			<channel id="moving" typeId="moving"/>
+			<channel id="command" typeId="command"/>
+			<channel id="t4command" typeId="t4command"/>
+		</channels>
+
+		<representation-property>id</representation-property>
+
+		<config-description>
+			<parameter name="id" type="text" required="true">
+				<label>ID</label>
+				<description>ID of the gate on the TP4 bus connected to the bridge.</description>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+
+	<channel-type id="doorstatus">
+		<item-type>String</item-type>
+		<label>Gate Status</label>
+		<state readOnly="true">
+			<options>
+				<option value="open">Open</option>
+				<option value="closed">Closed</option>
+				<option value="opening">Opening</option>
+				<option value="closing">Closing</option>
+				<option value="stopped">Stopped</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="moving">
+		<item-type>Switch</item-type>
+		<label>Moving</label>
+		<description>Indicates if the device is currently operating a command</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="obstruct">
+		<item-type>String</item-type>
+		<label>Obstruction</label>
+		<description>Something obstructs normal movement of the gate</description>
```

Comment:
What does it mean, can you give an example of such a string? "Something obstructs" seems to me could be evaluated to boolean (i.e. `Switch`), so probably the description is misleading me.

Reply:
The meaning of this value returned by the API is not completely clear for me neither - the name of the property is the only information I found to try to understand it - reason why I have choosen to keep it as a string.

Reply:
Let's see if your gate will be obstructed before the 4.0 release, so we can learn something. 🙂

Reply:
Ok. Got it. I did put my coat (it's raining) and obstructed by myself. It is a number (0 or 1) set when one crosses the infra-red barrer.